src/Compilers/Core/Portable/SourceGeneration/UserFunction.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// A Func based delegate that represents a user provided function
+    /// that has been wrapped to convert exceptions to UserCodeExceptions
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <typeparam name="TResult"></typeparam>
+    /// <param name="arg"></param>
+    /// <returns></returns>
+    internal delegate TResult UserFunc<in T, out TResult>(T arg);
```

Comment:
Why can't we use `Func<T, TResult>` here instead of declaring a new `delegate` type?